libraries/ArduinoOTA/ArduinoOTA.cpp

```diff
@@ -358,6 +358,27 @@ int ArduinoOTAClass::getCommand() {
   return _cmd;
 }
 
+// To avoid possible time-based attacks present function compares given 
+// strings in a constant time.
+bool ArduinoOTAClass::constantTimeEquals(const String & string1, 
+  const String & string2) {
+
+  // Preliminary check  
+  if(string1.length() != string2.length()) {
+	  return false;
+  }
+  
+  // Evaluates every character
+  bool equals = true;
+  unsigned int len = string1.length();
+  for(unsigned int i = 0; i < len; i++){
+    equals &= (string1.charAt(i) == string2.charAt(i));
```

Comment:
It would be interesting to see the generated code for this loop at -O2 optimization level... I wonder whether the compiler is smart enough to convert this 'and' into 'branch-if-zero'.

Reply:
I agree. Maybe better to write not so easily optimizable code. Ill think about that